c7n/resources/rds.py

```diff
@@ -1877,3 +1877,43 @@ def process(self, resources, event=None):
             if expected_dates.issubset(snapshot_dates):
                 results.append(r)
         return results
+
+
+@filters.register('engine')
+class EngineFilter(ValueFilter):
+    """
+    Filter a rds resource based on its Engine Metadata
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: find-deprecated-versions
+              resource: aws.rds
+              filters:
+                - type: engine
+                  key: Status
+                  value: deprecated
+    """
+
+    schema = type_schema('engine', rinherit=ValueFilter.schema)
+
+    permissions = ("rds:DescribeDBEngineVersions", )
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+        paginator = client.get_paginator('describe_db_engine_versions')
+        response = paginator.paginate(IncludeAll=True)
```

Comment:
rather than iterate all engine versions versions, its better I think to unique the set of engines/versions on the rds instances and just fetch those via describe_db_engine_versions with filters.

Reply:
yup make sense. done